lightning/src/chain/channelmonitor.rs

```diff
@@ -1425,39 +1502,49 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 				check_htlc_fails!(txid, "previous", 'prev_loop);
 			}
 
+			let htlc_claim_reqs = self.get_counterparty_htlc_output_claim_reqs(commitment_number, commitment_txid, Some(tx));
+			for req in htlc_claim_reqs {
+				claimable_outpoints.push(req);
+			}
+
+		}
+		(claimable_outpoints, (commitment_txid, watch_outputs))
+	}
+
+	fn get_counterparty_htlc_output_claim_reqs(&self, commitment_number: u64, commitment_txid: Txid, tx: Option<&Transaction>) -> Vec<ClaimRequest> {
+		let mut claims = Vec::new();
+		if let Some(htlc_outputs) = self.counterparty_claimable_outpoints.get(&commitment_txid) {
 			if let Some(revocation_points) = self.their_cur_revocation_points {
 				let revocation_point_option =
 					if revocation_points.0 == commitment_number { Some(&revocation_points.1) }
-					else if let Some(point) = revocation_points.2.as_ref() {
-						if revocation_points.0 == commitment_number + 1 { Some(point) } else { None }
-					} else { None };
+				else if let Some(point) = revocation_points.2.as_ref() {
```

Comment:
IMO it would be worthy to document this code:

```
diff --git a/lightning/src/chain/channelmonitor.rs b/lightning/src/chain/channelmonitor.rs
index 79c91d9b..1878362e 100644
--- a/lightning/src/chain/channelmonitor.rs
+++ b/lightning/src/chain/channelmonitor.rs
@@ -1516,10 +1516,16 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
                if let Some(htlc_outputs) = self.counterparty_claimable_outpoints.get(&commitment_txid) {
                        if let Some(revocation_points) = self.their_cur_revocation_points {
                                let revocation_point_option =
-                                       if revocation_points.0 == commitment_number { Some(&revocation_points.1) }
-                               else if let Some(point) = revocation_points.2.as_ref() {
-                                       if revocation_points.0 == commitment_number + 1 { Some(point) } else { None }
-                               } else { None };
+                                       // If counterparty commitment tx is the latest valid state, use their latest per-commitment point
+                                       if revocation_points.0 == commitment_number { 
+                                               Some(&revocation_points.1) 
+                                       } else if let Some(point) = revocation_points.2.as_ref() {
+                                               // If counterparty commitment tx is the previous latest valid state, use their previous latest per-commitment point
+                                               // (non-atomicity of revocation means it's lawful for them to have temporarily two valid commitment txn from our
+                                               // viewpoint)
+                                               if revocation_points.0 == commitment_number + 1 { Some(point) } else { None }
+                                       } else { None };
+
                                if let Some(revocation_point) = revocation_point_option {
                                        for (_, &(ref htlc, _)) in htlc_outputs.iter().enumerate() {
                                                if let Some(transaction_output_index) = htlc.transaction_output_index {
```

Can we really hit the None case or wouldn't this a hint of a more serious issue like lost monitor state ? Tweaking it as a panic we don't have coverage for this, and not even sure you can hit it even assuming state loss. Reaching it means you have at least some `per_commitment_option`.

Reply:
Docs is better, sure. I was just noting that the only change here was indentation - and I think the old indentation was much clearer than the new. Best to just not have any diff instead of changing unrelated code.

Reply:
As for hitting the `None` case, it should be? Code reads to me like the `None` case should be hit if our counterparty broadcasts a non-revoked state, but maybe I need to read deeper?

Reply:
Added the docs @ariard and fixed the ugly indentation